plugins/new-relic/retention.yaml

```diff
@@ -808,4 +808,189 @@ presetScripts:
+            kind=px.otel.trace.SPAN_KIND_CLIENT,
+            attributes={
+              'redis.req_bytes': otel_df.req_bytes,
+              'redis.resp_bytes': otel_df.resp_bytes,
+              'redis.resp_latency': otel_df.latency,
+              'redis.req_args': otel_df.req_args,
+              'redis.resp': otel_df.resp,
+            },
+          )
+        ]
+      ))
+  - name: Redis Metrics
+    defaultDisabled: true
+    defaultFrequencyS: 10
+    script: |
+      import px
+
+      def add_source_dest_columns(df):
+          df.pod = df.ctx['pod']
+          df.namespace = df.ctx['namespace']
+
+          # If remote_addr is a pod, get its name. If not, use IP address.
+          df.ra_pod = px.pod_id_to_pod_name(px.ip_to_pod_id(df.remote_addr))
+          df.is_ra_pod = df.ra_pod != ''
+          df.ra_name = px.select(df.is_ra_pod, df.ra_pod, df.remote_addr)
+
+          df.is_server_tracing = df.trace_role == 2
+          df.is_source_pod_type = px.select(df.is_server_tracing, df.is_ra_pod, True)
+          df.is_dest_pod_type = px.select(df.is_server_tracing, True, df.is_ra_pod)
 
+          # Set client and server based on trace_role.
+          df.source_pod = px.select(df.is_server_tracing, df.ra_name, df.pod)
+          df.destination_pod = px.select(df.is_server_tracing, df.pod, df.ra_name)
+
+          df = df.drop(['ra_pod', 'is_ra_pod', 'ra_name', 'is_server_tracing'])
+          df.source_service = px.pod_name_to_service_name(df.source_pod)
+          df.destination_service = px.pod_name_to_service_name(df.destination_pod)
+          # If the destination service is missing then try the nslookup
+          df.destination_service = px.select(
+              df.destination_service != '',
+              df.destination_service,
+              px.nslookup(df.remote_addr),
+          )
+          # If the destination service is still missing then set the remote_addr
+          df.destination_service = px.select(
+              df.destination_service != '',
+              df.destination_service,
+              df.remote_addr,
+          )
+          return df
+
+
+      df = px.DataFrame('redis_events', start_time=px.plugin.start_time, end_time=px.plugin.end_time)
+      df = add_source_dest_columns(df)
+
+      df.latency = df.latency / (1000 * 1000)
+      df.req_bytes = px.length(df.req_args)
+      df.resp_bytes = px.length(df.resp)
+      df = df.groupby(['req_cmd', 'source_pod', 'source_service', 'destination_pod', 'destination_service']).agg(
+          latency_count=('latency', px.count),
+          latency_min=('latency', px.min),
+          latency_max=('latency', px.max),
+          latency_sum=('latency', px.sum),
+          time_=('time_', px.max),
+          req_bytes=('req_bytes', px.sum),
+          resp_bytes=('resp_bytes', px.sum),
+      )
+
+      df.destination_namespace = px.pod_name_to_namespace(df.destination_pod)
+      df.source_namespace = px.pod_name_to_namespace(df.source_pod)
```

Comment:
Need to strip the namespace from Pixie's pod & service names:
```
df.source_namespace = px.pod_name_to_namespace(df.source_pod)
ns_prefix = df.source_namespace + '/'
df.source_pod = px.strip_prefix(ns_prefix, df.source_pod)
df.source_service = px.strip_prefix(ns_prefix, df.source_service)

df.destination_namespace = px.pod_name_to_namespace(df.destination_pod)
ns_prefix = df.destination_namespace + '/'
df.destination_pod = px.strip_prefix(ns_prefix, df.destination_pod)
df.destination_service = px.strip_prefix(ns_prefix, df.destination_service)
```

Reply:
this doesn't work unfortunately

Reply:
```
def remove_ns_prefix(column):
    return px.replace('[a-z0-9\-]*/', column, '')

df.pod = remove_ns_prefix(df.pod)
```